README.md

```diff
@@ -14,39 +14,53 @@ An overview of the architecture is shown below:
+| --target-grpc-port                 | int     | 50051            | gRPC port for warm up requests                                                                                  |
+| --target-http-host                 | string  | http://localhost | Http host to warm up                                                                                            |
+| --target-http-port                 | int     | 8080             | Http port for warm up requests                                                                                  |
+| --target-insecure                  | bool    | false            | Whether to skip TLS validation                                                                                  |
+| --target-readiness-path            | string  | /ready           | The path used for target readiness probe                                                                        |
+| --target-readiness-timeout-seconds | int     | -1               | Timeout for target readiness probe                                                                              |
+| --timeout-seconds                  | int     | 60               | Time after which warm up will stop making requests                                                              |
+
+### Warmup request
 A warmup request can be an HTTP one (over REST) or a gRPC one.
 
-#### HTTP (REST) requests
-HTTP requests are in the form `http:method:url:body` where `method` is one of `get`, `post`, or `put`, `url` is the url where the request will be sent, and `body` is a properly escaped JSON-formatted string.
+#### HTTP requests
 
-Indicative examples are shown below:
-- `http:get:/health`: HTTP GET request.
-- `http:post:/warmupUrl:{"key":"value"}`: POST request with its url being `/warmupUrl` and its body being `{"key":"value"}`.
-- `http:put:/warmupUrl:{"key":"value"}`: PUT request with its url being `/warmupUrl` its body being `{"key":"value"}`.
+HTTP requests are in the form `method:path:body` (`body` is optional).
+Host and port are taken from `--target-http-host` and
+`--target-http-port` flags.
+
+E.g.:
+ - `get:/health`: HTTP GET request.
+ - `post:/warmupUrl:{"key":"value"}`: POST request with its url being `/warmupUrl` and its body being `{"key":"value"}`.
 
 #### gRPC requests
-gRPC requests are in the form `grpc:service/method:message` where `service` and `method` are the names of the gRPC service and method respectively, and `message` is a properly escaped JSON-formatted string.
 
-_Note_: For both HTTP and gRPC requests you can use {today} and {today+n} to obtain the date for today or today +/- n days in YYYY-MM-DD format. For HTTP requests the date templating works for both URLs and body.
+gRPC requests are in the form `service/method:message`. Host and port are taken from `--target-grpc-host` and
```

Comment:
Although it's a good practice to always provide a message, even one with no fields, I believe an empty/null message is valid based on specs.
That said, can we change to `service/method[:message]`?